frontend/src/app/authentication/registration-dialog/registration-dialog.component.ts

```diff
@@ -0,0 +1,55 @@
+import {Component, OnInit} from '@angular/core';
+import {AuthenticationService} from "../../api/services";
+import {FormBuilder, FormControl, FormGroup, Validators} from "@angular/forms";
+
+@Component({
+    selector: 'app-registration-dialog',
+    templateUrl: './registration-dialog.component.html',
+    styleUrls: ['./registration-dialog.component.sass']
+})
+export class RegistrationDialogComponent implements OnInit {
+
+    public form: FormGroup;
+
+    constructor(public service: AuthenticationService, protected readonly formBuilder: FormBuilder) {
+        this.form = this.formBuilder.group({
+            name: new FormControl('', Validators.required),
+            email: new FormControl('', [Validators.required, Validators.email]),
+            password: new FormControl('', [Validators.required, Validators.minLength(16), Validators.maxLength(64)]),
+            passwordConfirmation: new FormControl('', [Validators.required])
+        }, );
+    }
+
+    ngOnInit(): void {
+    }
+
+    submit() {
+        if (this.isFormValid()) {
+            this.service.createUser({
+                body: {
+                    name: this.form.get('name')?.value,
+                    email: this.form.get('email')?.value,
+                    password: this.form.get('password')?.value
+                }
+            }).subscribe(next => console.log(next), error => console.log(error))
```

Comment:
These `console.log` calls should probably be replaced with proper error handling